lib/transfer.c

```diff
@@ -200,10 +200,10 @@ CURLcode Curl_fillreadbuffer(struct connectdata *conn, size_t bytes,
     else {
       failf(data, "operation aborted by trailing headers callback");
       *nreadp = 0;
+      Curl_add_buffer_free(&data->state.trailers_buf);
```

Comment:
The former way seems more correct, since it's effectively the cleanup block. Also iirc that function is idempotent?

Reply:
The former way would lead to a double-free if the `Curl_add_bufferf` call within `Curl_http_compile_trailers` failed.

To make your fix work, I suppose we should made `Curl_http_compile_trailers` clear the `data->state.trailers_buf` pointer on error. Which certainly could be a good idea...